src/Tasks/GenerateResource.cs

```diff
@@ -1969,9 +1975,14 @@ private bool NeedSeparateAppDomainBasedOnSerializedType(XmlReader reader)
         /// </summary>
         private bool DetermineWhetherSerializedObjectLoads(string data)
         {
+            if (WarnOnBinaryFormatterUse)
+            {
+                Log.LogWarningWithCodeFromResources("GenerateResource.BinaryFormatterUse");
```

Comment:
This is not the right place to put the warning. What we want to warn about is not build-time use of BinaryFormatter, which is already avoidable with `UsePreserializedResources`, but the fact of embedding a BinaryFormatter resource, which wouldn't be noticed (in .NET/`UsePreserializedResources` mode) on this codepath.